samples/cellular/modem_shell/src/gnss/gnss.c

```diff
@@ -113,7 +119,8 @@ static struct k_work get_pgps_data_work;
 	!defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS)) || \
 	(defined(CONFIG_NRF_CLOUD_PGPS) && \
 	 !defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS))) && \
-	!defined(CONFIG_NRF_CLOUD_MQTT)
+	!defined(CONFIG_NRF_CLOUD_MQTT) && \
+	defined(CONFIG_NRF_CLOUD_REST)
```

Comment:
I think this boolean logic needs to be split up and given some explanatory comments, maybe in followup PR. For instance it isn't clear from this why enabling MQTT should disable specifically the JWT buffer

Reply:
I'll leave that for @tokangas.

Reply:
@glarsennordic I agree that this is ugly. The complexity comes from the fact that REST is not always used to fetch assistance data even though `CONFIG_NRF_CLOUD_REST` is enabled. For example, if LwM2M is enabled, it's preferred over REST and without this nasty flagging, a compilation warning would be generated for unused REST buffers. The same happens also if both `CONFIG_NRF_CLOUD_MQTT` and `CONFIG_NRF_CLOUD_REST` are enabled. The purpose of this flagging is to enable REST buffers only when REST is used, so we don't get compilation warnings with any of the supported configurations.

It would be simpler if only one transport could be enabled at a time, but currently mosh supports some combinations. There was even some discussion about adding even more supported combinations, but that would increase the complexity even more. I'm not sure what is the point, because for example if both MQTT and REST are enabled, I think that everything goes over MQTT anyway, so REST support can't be used for much. Probably to update the device shadow or perform JITP, but I think that's all.

Then again, if such combinations are supported by our libraries and make sense on some level, then it would be good to also test those.

If you like, I could add a comment to explain what the flagging does and why it's necessary.

Reply:
Yes, comments would be a good addition.

Beyond comments, I was also suggesting that you could split the boolean logic into expository statements.

I haven't tested, but I think this should work:

```
#define AGPS_USES_REST  (defined(CONFIG_NRF_CLOUD_REST) && \ 
			!defined(CONFIG_NRF_CLOUD_MQTT) && \
			 defined(CONFIG_NRF_CLOUD_AGPS) && \
			!defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS))
#define PGPS_USES_REST  (defined(CONFIG_NRF_CLOUD_REST) && \ 
			!defined(CONFIG_NRF_CLOUD_MQTT) && \
			 defined(CONFIG_NRF_CLOUD_PGPS) && \
			!defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS))
	
#if (AGPS_USES_REST || PGPS_USES_REST)
/* JWT and recieve buffers will be needed if either AGPS or PGPS uses REST */
static char jwt_buf[600];
static char rx_buf[2048];
#endif
```

Reply:
@glarsennordic Your proposal caused compilation warnings, so I had to change it a bit. Created PR #11875, let's see if all configurations can still be built without warnings.